app/src/main/java/org/thoughtcrime/securesms/database/RecipientSettingsDatabase.kt

```diff
@@ -143,6 +145,88 @@ class RecipientSettingsDatabase @Inject constructor(
         }
     }
 
+    /**
+     * This method returns all profile pic url and key.
+     * This will be used to identify which avatars are still being used  to exclude
+     * them from the cleanup.
+     */
+    fun getAllReferencedAvatarFiles(): Set<RemoteFile.Encrypted> {
+        val recipientAvatars = LinkedHashSet<RemoteFile.Encrypted>()
+        readableDatabase.rawQuery(
+            """
+        SELECT $COL_PROFILE_PIC_URL, $COL_PROFILE_PIC_KEY
```

Comment:
Should we use DISTINCT?
```suggestion
        SELECT DISTINCT $COL_PROFILE_PIC_URL, $COL_PROFILE_PIC_KEY
```